### ISSUE TITLE
Weapon / Action Library

### ISSUE THREAD

**Issue by tdhsmith** [cynicaloptimist/improved-initiative#413]
Abstract weapons out to their own library, so that a character can simply reference a weapon in the library and quickly link in that weapon's action text (or more complexly, import _multiple_ actions from a single weapon e.g. for magic weapons with more than one ability).

**Comment by cynicaloptimist** [comment]
Ambitious idea. This should be more general than weapons, something like a "Feature" library that could include weapons, feats, etc.

**Comment by tdhsmith** [comment]
Yeah I suppose you're right; it may as well support using any "template" element.

I think the current questions for me are:
- Can the user attach a feature and then override the values it imports? _I feel this adds unneeded complexity when they could instead browse to its library entry and manually copy-paste if they want to create a modified version_
- Do "updates" from the feature occur after initial attachment? _Library items aren't likely to change often, so it doesn't feel necessary._
- Is there a separate UI for attaching features or are their names automatically detected in labels & notes? _I could go either way here. I think even if they were auto-detected, you'd still want a yes/no dialog to verify, lest people be unable to use certain strings without triggering the feature._
- Can users create their own features? _Eventually for sure, but I don't necessarily think v1 needs to include it._
- Should the combatant UI indicate when an effect comes from a feature?
- Can features add multiple kinds of effects? _This seems necessary. For example, things like magical armor are going to provide several of AC, skills, saves, and actions. I think the only viable alternative would be to instead implement some items as multiple separate templates, but that seems like a pretty big usability loss.

**Comment by tdhsmith** [comment]
This opens the question of whether you were intending to eventually support combatant **inventories** too, since item-like features would most naturally live in an inventory menu.

I guess the stat block editor could just have a generic section for all feature library items though, so it's not something that needs to be decided immediately.
